src/main/webapp/app/entities/exercise-category.model.ts

```diff
@@ -1,6 +1,30 @@
 export class ExerciseCategory {
     public color?: string;
+
+    // TODO should be renamed to "name" -> accessing variable via "category.name" instead of "category.category" - requires database migration (stored as json in database, see the table "exercise_categories")
     public category?: string;
 
-    constructor() {}
+    constructor(color: string | undefined, category: string | undefined) {
```

Comment:
Would this work?

```suggestion
    constructor(color?: string, category?: string) {
```

Reply:
It would work, but would change the semantics of this method. `: string | undefined` requires a parameter that might also be undefined `?: string` is an optional parameter that may be undefined or can even be left out when calling this method

Reply:
I defined it like that on purpose to prevent accidentally only the color is set and always bought parameters need to be defined - I just changed the order of the category and color in the constructor to make it less error-prone if it is changed in the future and the color is left out

I do slightly prefer the current approach